tournaments/round_robin_tournament.py

```diff
@@ -0,0 +1,51 @@
+from core.connect_four_game import ConnectFourGame
+from core.board import Board
+from itertools import permutations
+from functools import reduce
+
+class RoundRobinTournament:
+    def __init__(self, players, view):
+        self.players = players
+        self.matches = []
+        self.results = {}
+        self.view = view
+    
+    def _calc_match_permutations(self):
+        num_players = len(self.players)        
+        for player in range(num_players):
+            other_players = list(range(num_players))
+            other_players.remove(player)
+            [self.matches.append([player, opponent]) for opponent in other_players if not self._match_exists(player, opponent)]        
+    
+    def _match_exists(self, player1_index, player2_index):
+        for match in self.matches:
+            if player1_index in match and player2_index in match:
+                return True
+        return False
+    
+    def _clear_results(self):
+        self.results.clear()
+        for player in self.players:
+            self.results[player] = []
+
+    def _run_off_matches(self):
+        self._clear_results()
+        for match in self.matches:
+            game = ConnectFourGame(Board(width=7, height=6))
```

Comment:
In the future, I'd like to have the `Game` passed into the tournament. The tournament would only care about matching multiple players together and then using the results of a `Game` to determine a ranking. Then we could extend game to add rules like best of 3 or first to 10 or loser goes first etc etc

Before this happens though I think we'd need the output of game and have it be something more flexible.

Reply:
Yeah, that does make more sense. I just wanted to get this ready for tomorrow haha.